File: src/types/employee.ts
export type EmployeeStatus = 'Active' | 'Inactive' | 'OnLeave' | 'Terminated';
export type Department = 'FrontDesk' | 'Housekeeping' | 'Maintenance' | 'FoodAndBeverage' | 'Management' | 'Other';
export type Position = 'Manager' | 'Supervisor' | 'Staff' | 'Trainee' | 'Intern';

export interface Employee {
    id: number;
    employeeId: string;
    name: string;
    position: Position;
    department: Department;
    email: string;
    phone: string;
    startDate: string;
    status: EmployeeStatus;
    schedule?: string;
    createdAt: string;
    lastUpdated?: string;
}

export interface EmployeeCreate {
    employeeId: string;
    name: string;
    position: Position;
    department: Department;
    email: string;
    phone: string;
    startDate: string;
    schedule?: string;
}

export type EmployeeUpdate = Partial<EmployeeCreate> & {
    status?: EmployeeStatus;
}; 

File: SmartStayHaven.API/Models/Employee.cs
using System.ComponentModel.DataAnnotations;

namespace SmartStayHaven.API.Models
{
    public class Employee
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string EmployeeId { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Position { get; set; }

        [Required]
        public string Department { get; set; }

        [Required]
        [EmailAddress]
        public string Email { get; set; }

        [Required]
        [Phone]
        public string Phone { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public string Status { get; set; } = "Active";

        public string? Schedule { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? LastUpdated { get; set; }
    }
} 